crates/bevy_ptr/src/lib.rs

```diff
@@ -357,3 +374,36 @@ impl<'a, T> UnsafeCellDeref<'a, T> for &'a UnsafeCell<T> {
         self.get().read()
     }
 }
+
+trait EnsureAligned {
+    fn ensure_aligned(self) -> Self;
+}
+
+// Disable this for miri runs as it already checks if pointer to reference
+// casts are properly aligned.
+#[cfg(all(debug_assertions, not(miri)))]
+impl<T: Sized> EnsureAligned for *mut T {
+    #[inline(always)]
+    fn ensure_aligned(self) -> Self {
+        let align = core::mem::align_of::<T>();
+        // Implemenation shamelessly borrowed from the currently unstable
+        // ptr.is_aligned_to.
+        //
+        // Replace once https://github.com/rust-lang/rust/issues/96284 is stable.
+        assert!(
+            self as usize & (align - 1) == 0,
+            "pointer is not aligned. Address {:p} does not have alignemnt {}",
```

Comment:
Is there any way of adding the type name to this error message without adding `'static` bounds everywhere?

Reply:
[`std::any::type_name`(https://doc.rust-lang.org/stable/std/any/fn.type_name.html)] does not require static?